### PR TITLE
Csharp support

### PR DESCRIPTION
**Pull request by benjstephenson**
Added a csharp module with treesitter, LSP via omnisharp and debug via notecoredbg.
For the debugger, I'm not quite sure how to get the install path of netcoredbg; I've just assumed `/usr/local/bin/netcoredbg` so happy to take some direction on how to make that nicer.

### CODE REVIEW

File: fnl/modules/tools/debugger/config.fnl [hunk 1]
```diff
@@ -24,11 +24,27 @@
...
+                                 (.. (vim.fn.getcwd) "/bin/Debug/" :file)))
+       }])
+
 (nyoom-module-p! cc
                  (do
                    (set dap.configurations.c lldb-configs)
                    (set dap.configurations.c dap.configurations.cpp)))
 
+(nyoom-module-p! csharp (set dap.configurations.cs coreclr-configs))
+
```
Comment (hbjydev) [inline fnl/modules/tools/debugger/config.fnl:47]: should this not be moved to a `(do [...])` block inside the `nyoom-module-p!` call for the `csharp` module?

```suggestion
(local coreclr-configs
       [{:name "netcoredbg"
        :type :coreclr
        :request :launch
        :program (fn []
                   (vim.fn.input "Path to executable: "
                                 (.. (vim.fn.getcwd) "/bin/Debug/" :file)))
       }])

(nyoom-module-p! cc
                 (do
                   (set dap.configurations.c lldb-configs)
                   (set dap.configurations.c dap.configurations.cpp)))

(nyoom-module-p! csharp (do
                                                   (set dap.configurations.cs coreclr-configs)
                                                   (set dap.adapters.coreclr
                                                        {:type :executable
                                                        :command "/usr/local/bin/netcoredbg/netcoredbg"
                                                        :args ["--interpreter=vscode"]})))
```

**Review by hbjydev — COMMENTED** [review]
Thought on this :)

File: fnl/modules/tools/debugger/config.fnl (hunk 1 above)
Comment (shaunsingh) [reply fnl/modules/tools/debugger/config.fnl:47]: For now I'd prefer to keep all of the debugger stuff in the debugger module, it's what doom does. Not sure if I'll keep it in one file in the future, but for now its easier to work with (and keeps merge conflicts to a minimum)

Comment (shaunsingh) [reply fnl/modules/tools/debugger/config.fnl:47]: Oh my mistake, I read it incorrectly. Indeed it should be moved to a `do` block

Comment (benjstephenson) [reply fnl/modules/tools/debugger/config.fnl:47]: Great, I'll update the PR today. Thanks for the feedback!

Comment (benjstephenson) [reply fnl/modules/tools/debugger/config.fnl]: Pushed an update @hbjydev, do you want me to resolve this or should it be you? Not sure if there's a policy around that or not is all.

Comment (shaunsingh) [reply fnl/modules/tools/debugger/config.fnl]: No policy, looks good to me. thanks!

**Pull request merged**